tests/src/test/java/alluxio/client/rest/S3ClientRestApiTest.java

```diff
@@ -616,6 +635,151 @@ public void listBucketZeroMaxKeys() throws Exception {
+        .runAndCheckResult(expected);
+
+    //parameters with prefix=""
+    expected = new ListBucketResult("bucket", statuses,
+        ListBucketOptions.defaults().setPrefix(""));
+    assertEquals(3, expected.getContents().size());
+    assertEquals("file0", expected.getContents().get(0).getKey());
+    assertEquals("file_store/", expected.getContents().get(1).getKey());
+    assertEquals("file_store/file1", expected.getContents().get(2).getKey());
+
+    parameters.put("prefix", "");
+    new TestCase(mHostname, mPort, mBaseUri,
+        "bucket", parameters, HttpMethod.GET,
+        TestCaseOptions.defaults().setContentType(TestCaseOptions.XML_CONTENT_TYPE))
+        .runAndCheckResult(expected);
+
+    //parameters with non-existent prefix="foo"
+    try {
+      expected = new ListBucketResult("bucket", statuses,
+          ListBucketOptions.defaults().setPrefix("foo"));
+    } catch (Exception e) {
+      // expected
+      // TODO(czhu): with the current implementation of prefixes w/o delimiters,
+      // this is never an error because we just list the entire bucket recursively
+      statuses = new ArrayList<>();
+      return;
+    }
+    assertEquals(0, expected.getContents().size());
+
+    parameters.put("prefix", "foo");
+    new TestCase(mHostname, mPort, mBaseUri,
+        "bucket", parameters, HttpMethod.GET,
+        TestCaseOptions.defaults().setContentType(TestCaseOptions.XML_CONTENT_TYPE))
+        .runAndCheckResult(expected);
+
+    // prefix test with delimiter="/"
+    statuses = mFileSystem.listStatus(new AlluxioURI("/bucket"),
+        ListStatusPOptions.newBuilder().setRecursive(false).build());
+
+    //parameters with prefix="file"
+    expected = new ListBucketResult("bucket", statuses,
+        ListBucketOptions.defaults().setPrefix("file").setDelimiter(AlluxioURI.SEPARATOR));
+    assertEquals(1, expected.getContents().size());
+    assertEquals("file0", expected.getContents().get(0).getKey());
+    assertEquals(1, expected.getCommonPrefixes().size());
+    assertEquals("file_store/", expected.getCommonPrefixes().get(0).getPrefix());
+
+    parameters.put("delimiter", AlluxioURI.SEPARATOR);
+    parameters.put("prefix", "file");
+    new TestCase(mHostname, mPort, mBaseUri,
+        "bucket", parameters, HttpMethod.GET,
+        TestCaseOptions.defaults().setContentType(TestCaseOptions.XML_CONTENT_TYPE))
+        .runAndCheckResult(expected);
+
+    //parameters with prefix="file_store/file1/"
+    try {
+      statuses = mFileSystem.listStatus(new AlluxioURI("/bucket/file_store/file1/"),
+          ListStatusPOptions.newBuilder().setRecursive(false).build());
+    } catch (Exception e) {
+      // expected
+      // TODO(czhu): with the current implementation of listStatus() the trailing '/' character
+      // doesn't cause a failure, despite `/bucket/file_store/file1` not being a directory
+      statuses = new ArrayList<>();
+      return;
+    }
```

Comment:
FYI is the Alluxio gRPC `ListStatus` supposed to return the object itself if the provided path is not a directory?

@yuzhu PTAL

Reply:
if a user specifies a slash at the end of the path, the name resolution should have failed at the alluxio level and never reach the ufs level, I don't see why we are not returning an exception.  I would not change the behavior in this PR, but I don't think it is the right behavior either.

Reply:
It looks like the constructor of AlluxioURI will always remove a / at the end of a string